derp/derp_server.go

```diff
@@ -215,6 +225,18 @@ func (s *Server) PrivateKey() key.Private { return s.privateKey }
 // PublicKey returns the server's public key.
 func (s *Server) PublicKey() key.Public { return s.publicKey }
 
+// Returns a (possibly-reused) byte slice of the given length
+func (s *Server) empty_bytes(length uint32) []byte {
+	zeroed := s.bytes_pool.Get().([]byte)
+	for i := range zeroed {
+		zeroed[i] = 0
+	}
+	if uint32(cap(zeroed)) >= length {
+		return zeroed[:length]
+	}
+	return append(zeroed, make([]byte, length-uint32(len(zeroed)))...)
```

Comment:
this append then throws away the old buffer.

better to put it back and allocate a new one the right size.

but best is to not have one pool for all possible size classes. In the standard library see the bits like:

```
src/net/http/server.go: bufioReaderPool   sync.Pool
src/net/http/server.go: bufioWriter2kPool sync.Pool
src/net/http/server.go: bufioWriter4kPool sync.Pool
```

For even fancier things people do, see e.g. https://github.com/golang/go/issues/27735#issuecomment-739169121

Reply:
for now I'm just going to put it back, and leave possible size classes for future optimizations